yew/src/dev/mod.rs

```diff
@@ -0,0 +1,228 @@
+//! Developer tools.
+//! These communicate with a debugging server (normally a browser extension) over a WebSocket connection.
+
+use serde::Serialize;
+use serde_json;
+
+#[cfg(test)]
+pub use messages::tests as message_tests;
+
+#[cfg(feature = "web_sys")]
+use web_sys;
+
+#[cfg(feature = "std_web")]
+use stdweb;
+
+pub mod messages;
+
+/// Stores a connection to the DevTools server.
+#[derive(Debug, Clone)]
+pub struct DebuggerConnection {
+    #[cfg(feature = "web_sys")]
+    /// Public only for testing.
+    pub ws: web_sys::WebSocket,
+    #[cfg(feature = "std_web")]
+    /// Public only for testing.
+    pub ws: stdweb::web::WebSocket,
+    message_queue: Vec<String>,
+}
+
+/// Describes the state of a WebSocket connection.
+#[derive(Debug)]
+pub enum ConnectionState {
```

Comment:
Is this used?

Reply:
No (and the future isn't being used either). It's some code I had here when I was trying to work out how to handle the WebSocket connection.
I'm now using an event listener to send messages when the connection opens (I'll also add one to raise an error if it can't connect).